micro-rdk/src/common/conn/viam.rs

```diff
@@ -553,22 +555,29 @@ where
         let robot = Arc::new(Mutex::new(robot));
 
         if self.http2_server.has_http2_server() && !self.http2_server_insecure {
-            // TODO implement certificate storage
-            // in the situation where we are not connected to app or retrieving certs fails AND there are
-            // no certs stored then HTTP2 **should** be disabled.
-            let mut certs = None;
+            // Try to obtain and store a fresh TLS certificate. If this fails or we cannot reach
+            // app, then we'll end up falling back on whatever TLS certificate was cached
             if let Some(app) = app_client.as_ref() {
-                certs = app.get_certificates().await.ok();
+                if let Ok(cert_resp) = app.get_certificates().await {
```

Comment:
Can we change the logic so that we only load certs from the nvs if the app client is absent? If storing the cert failed but the response passed then we should be able to continue regardless.